nvtabular/loader/tensorflow.py

```diff
@@ -238,6 +240,7 @@ def __len__(self):
         """
         # TODO: what's a better way to do this inheritance
         # of the appropriate methods? A Metaclass?
+        DataLoader.stop(self)
```

Comment:
It seems a little bit weird to stop the dataloader when computing the length, in the sense that it's an unexpected side effect happening inside a method that's essentially a property of the object. Do TF training loops make a new iter each epoch? Wondering if [the stop inside the `__iter__` method](https://github.com/NVIDIA/NVTabular/blob/2a37c581f5807714732d8ab17245f583dfc1d9e2/nvtabular/loader/backend.py#L245) is sufficient? Or maybe there's another place we can put this?

Reply:
That does not work because the keras sequence class we are using only employs the len and get_item. We are doing this to eliminate the need for the infinite iterator behavior currently employed. https://www.tensorflow.org/api_docs/python/tf/keras/utils/Sequence. Next step would be to change this to a tf dataset instead of a keras sequence, since it is supported now https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit.

Reply:
Okay, cool! I'll write a ticket about that and stick it in a future release.